app/src/main/java/com/lagradost/cloudstream3/utils/AppUtils.kt

```diff
@@ -429,25 +430,39 @@ object AppUtils {
         }
     }
 
+    fun Activity.addRepositoryDialog(repositoryName: String, isExtensionsFragment: Boolean) {
+        val repos = RepositoryManager.getRepositories()
 
-    fun Activity.downloadAllPluginsDialog(repositoryUrl: String, repositoryName: String) {
-        runOnUiThread {
-            val context = this
-            val builder: AlertDialog.Builder = AlertDialog.Builder(this)
-            builder.setTitle(
-                repositoryName
-            )
-            builder.setMessage(
-                R.string.download_all_plugins_from_repo
-            )
-            builder.apply {
-                setPositiveButton(R.string.download) { _, _ ->
-                    downloadAll(context, repositoryUrl, null)
+        // navigate to newly added repository on pressing Open Repository
+        fun openAddedRepo() {
+
+            // don't redirect if user is adding manually from add repository fab button
+            if (!isExtensionsFragment && repos.isNotEmpty()) {
+                normalSafeApiCall {
+                    navigate(
+                    R.id.navigation_home_to_navigation_settings_plugins,
+                    PluginsFragment.newInstance(
+                        repositoryName,
+                        repos.last().url,
+                        false)
+                    )
                 }
+            }
+        }
 
-                setNegativeButton(R.string.no) { _, _ -> }
+        runOnUiThread {
+            AlertDialog.Builder(this).apply {
+                setTitle(repositoryName)
+                setMessage(R.string.download_all_plugins_from_repo)
+                // won't show "open button" when adding from settings
+                if (!isExtensionsFragment) {
```

Comment:
Why??????

Reply:
Nah, removing that option just makes it a waste of a popup then

Reply:
what to do then, remove popup here or add button ? without popup we can show a toast of added successfuly, non intrusive also

Reply:
Users will just be confused it you "download" a repo, but the extentions does not get downloaded nor do they get redirected or even get a dialog about it. The dialog *should* have the option to just open it, because it is not obvious that more needs downloading, nor that it is clickable.